proposals/lip-0045.md

```diff
@@ -1575,9 +1492,82 @@ genesisInteroperabilityStoreSchema = {
+  * `activeValidators` must have at least 1 element and at most `MAX_NUM_VALIDATORS` elements;
+  * `activeValidators` must be ordered lexicographically by `blsKey` property;
+  * all `blsKey` properties must be pairwise distinct;  
+  * let `totalWeight` be the sum of the `bftWeight` property of every element in `activeValidators`. Then `totalWeight` has to be less than or equal to `MAX_UINT64`;
+  * check that `totalWeight//3 + 1 <= certificateThreshold <= totalWeight`, where `//` indicates integer division;
+  * check that the corresponding `validatorsHash` stored in `chainInfo.chainData.lastCertificate.validatorsHash` matches with the value computed from `activeValidators` and `certificateThreshold`.
+* For each entry `chainInfo` in `chainInfos`, `chainInfo.chainData.status == CHAIN_STATUS_TERMINATED` if and only if a corresponding entry (i.e., with `chainID == chainInfo.chainID`) exists in `terminatedStateAccounts`.
+* Each entry `stateAccount` in `terminatedStateAccounts` has a unique `stateAccount.chainID` and is ordered lexicographically by `stateAccount.chainID`.
+* For each entry `stateAccount` in `terminatedStateAccounts` holds `stateAccount.stateRoot == chainData.lastCertificate.stateRoot`, `stateAccount.mainchainStateRoot == EMPTY_HASH`, and `stateAccount.initialized == True`. Here `chainData` is the corresponding entry (i.e., with `chainID == stateAccount.chainID`) in `chainInfos`.
+* Each entry `outboxAccount` in `terminatedOutboxAccounts` has a unique `outboxAccount.chainID` and is ordered lexicographically by `outboxAccount.chainID`. Furthermore, an entry `outboxAccount` in `terminatedOutboxAccounts` must have a corresponding entry (i.e., with `chainID == outboxAccount.chainID`) in `terminatedStateAccounts`. Notice that the opposite is not necessarily true, so that there could be an entry in `terminatedStateAccounts` without a corresponding entry in `terminatedOutboxAccounts`.
+
+
+On a sidechain, the Interoperability state can only contain the chain account for the mainchain (if the mainchain registration was done) or be empty. Hence, `chainInfos` is either empty or it contains exactly one entry for the mainchain.
+
+If `chainInfos` is empty, then check that: 
+
+* `ownChainName` is the empty string;
+* `ownChainNonce == 0`;
+* `terminatedStateAccounts` is empty;
+* `terminatedOutboxAccounts` is empty.
+
+
+If `chainInfos` is not empty, then check that:
+
+* `ownChainName` is from the character set `a-z0-9!@$&_.`, has length between `MIN_CHAIN_NAME_LENGTH` and `MAX_CHAIN_NAME_LENGTH`, and `ownChainName != CHAIN_NAME_MAINCHAIN`;
+* `ownChainNonce > 0`;
+* `chainInfos` contains exactly one entry with:
+  * `chainID == getMainchainID()`;
+  * `chainData.name ==  CHAIN_NAME_MAINCHAIN`, `chainData.status !=  CHAIN_STATUS_TERMINATED`, and `chainData.lastCertificate.timestamp < g.header.timestamp`;
+  * `channelData.messageFeeTokenID == Token.getTokenIDLSK()`.
```

Comment:
Also here, we should enforce `minReturnFeePerByte == MIN_RETURN_FEE_PER_BYTE_LSK`.

Reply:
added